tests/components/alexa/test_intent.py

```diff
@@ -490,8 +490,11 @@ async def test_intent_session_ended_request(alexa_client):
 
     req = await _intent_req(alexa_client, data)
     assert req.status == HTTPStatus.OK
-    text = await req.text()
-    assert text == ""
+    data = await req.json()
+    assert (
+        data["response"]["outputSpeech"]["text"]
+        == "This intent is not yet configured within Home Assistant."
+    )
```

Comment:
We get this response because `amzn1.echo-sdk-ams.app.000000-d0ed-0000-ad00-000000d00ebe.SessionEndedRequest` is not a supported intent.

According to the docs there shouldn't be any response: https://developer.amazon.com/en-US/docs/alexa/custom-skills/request-types-reference.html#valid-response-types-2

Are the docs wrong, and this a useful response that we want to send back?
Is it correct to interpret a `SessionEndedRequest` as an intent, or should it have its own handler?

Reply:
I just tried to follow the pattern as there is an **special intent** with the `application id` to capture [LaunchRequest](https://github.com/home-assistant/core/blob/1d68934ae25821851ec7ed70e106a398337fb7a6/homeassistant/components/alexa/intent.py#L150), and not just the string `LaunchRequest`.
https://www.home-assistant.io/integrations/alexa.intent/#support-for-launch-requests

So I just followed the pattern for [SessionEndedRequest](https://github.com/home-assistant/core/commit/3146c001e83e16b0c99eab31078e167f42a553d2#diff-6a4361e5c0693eb9fb08193a576fd791f770af510479615982aaf8bc6e16fecdR149)... `application id + "." + type`
Also, I thought it might be possible to handle multiple application ids in the future.
That is the reason why I didn't just use the string `SessionEndedRequest` as the intent.

Capturing `SessionEndedRequest` is useful to let us define a default action when there isn't any response in time, as the intent may be composed of a list of actions.
https://www.home-assistant.io/integrations/intent_script/#action

Just for reference: https://github.com/keatontaylor/alexa-actions/blob/ae3d9e43983462faacd11df31a3079db4f2201ea/lambda/lambda_function.py#L322
The example above returns a pseudo intent called "ResponseNone".

Currently, I have the following generic code:
https://gist.github.com/w35l3y/fa7086172c441fdaec9cd1f0d4f72e36
| Intent | Behavior | Status |
|---|---|---|
| LaunchRequest | Sends `speech` and `reprompt` based on the content of a `input_text` | Working |
| AMAZON.YesIntent | Sends `speech` and emits an event with positive response to be captured by an automation. | Working |
| AMAZON.NoIntent | Sends `speech` and emits an event with negative response to be captured by an automation. | Working |
| SessionEndedRequest | Emits an event with fallback response based on the content of a `input_text` to be captured by an automation. | Waiting this PR |

Reply:
Oh, do you mean the purpose is to make it possible to handle `SessionEndedRequest` in scripts?

If that's the case, please open a documentation PR which adds an example to https://www.home-assistant.io/integrations/alexa.intent.
I also think a test case asserting that such a script is working, instead of just a test case for when there's no handler.

Reply:
The purpose is to fix the current response that is "There was a problem with the requested skill's response."
Instead, it will return "This intent is not yet configured within Home Assistant."

English is not my native language, but I will try my best to write something good this weekend, ok?

Reply:
> it will return "This intent is not yet configured within Home Assistant."

OK so the Alexa docs is misleading, it's actually possible to send a response back?
But my understanding is correct in that with your PR it's possible to handle `SessionEndedRequest` in a script/automation, but without your PR it's not possible?

> English is not my native language, but I will try my best to write something good this weekend

Yeah, that's perfectly fine, someone will help with the language/grammar if needed 👍

Reply:
Yes, that is right.
According to Alexa docs, it is possible to handle `SessionEndedRequest`.

For reference:
[Request Types Reference - SessionEndedRequest](https://developer.amazon.com/en-US/docs/alexa/custom-skills/request-types-reference.html#sessionendedrequest)
[Developing Your First Skill - SessionEndedRequest handler](https://developer.amazon.com/en-US/docs/alexa/alexa-skills-kit-sdk-for-python/develop-your-first-skill.html#sessionendedrequest-handler)

Before this PR, when Alexa is waiting some voice response and nothing is said in time (+- 5 to 8 seconds), it sends `SessionEndedRequest` to Home Assistant that returns [None](https://github.com/home-assistant/core/blob/692a602aeabc2f11dfed9bdb7d6ac2acd20bde8d/homeassistant/components/alexa/intent.py#L132), an error message ("There was a problem with the requested skill's response.") and doesn't allow us to handle the request.
This PR allows us to handle it properly.

Reply:
Added an example to the docs: https://github.com/home-assistant/home-assistant.io/pull/22908

Do you consider it resolved?